public/app/plugins/datasource/loki/modifyQuery.ts

```diff
@@ -397,3 +413,19 @@ function labelExists(labels: QueryBuilderLabelFilter[], filter: QueryBuilderLabe
 function findLastPosition(positions: Position[]): Position {
   return positions.reduce((prev, current) => (prev.to > current.to ? prev : current));
 }
+
+function getAllPositionsInNodeByType(query: string, node: SyntaxNode, type: number): Position[] {
+  if (node.type.id === Matcher) {
```

Comment:
I am curious, why do we need this exact `if` statement?

Reply:
Nice catch! This is supposed to be `if (node.type.id === type)` to catch cases where you look for  for example `Matcher` and your node is `Matcher`.  I'll add fix.

Reply:
Fixed in https://github.com/grafana/grafana/pull/57280/commits/29369bdd8d2a6a8af1418dfc690b7812ee56605e